### ISSUE TITLE
Unable to update config.yml in Rasa X

### ISSUE THREAD

**Issue by kearnsw** [RasaHQ/rasa#7832]
**Rasa version**:
2.2.7-full

**Rasa SDK version** (if used & relevant):
2.2.0

**Rasa X version** (if used & relevant):
0.35.0

**Python version**:
3.8

**Operating system** (windows, osx, ...):
Amazon Linux

**Issue**:
I am not able to update the configuration file. Each time I attempt to do so, I receive a 422 error. In addition, syncing with Github also does not appear to modify the config file. This error still occurs after disconnecting from Github. It is present on this version and also the 0.33.0 version used before. 

**Error (including full traceback)**:
```
[2021-01-28 04:20:44 +0000] - (sanic.access)[INFO][192.168.41.156:56032]: PUT http://<redacted>.us-east-1.elb.amazonaws.com/api/projects/default/settings  422 237
DEBUG:rasax.community.services.model_service:Model discovery completed.
```

**Command or request that led to error**:
Manually updating the configuration file by adding the following line:
```
- name: aws.ComprehendMedical
```

Also occurs by just removing anything or any other changes. 

**Content of configuration file (config.yml)** (if relevant):
```yml
language: en
pipeline:
  - name: WhitespaceTokenizer
  - name: RegexFeaturizer
  - name: LexicalSyntacticFeaturizer
  - name: CountVectorsFeaturizer
  - name: CountVectorsFeaturizer
    analyzer: char_wb
    min_ngram: 1
    max_ngram: 4
  - name: aws.ComprehendMedical
  - name: DIETClassifier
    epochs: 100
  - name: EntitySynonymMapper
  - name: ResponseSelector
    epochs: 100
  - name: FallbackClassifier
    threshold: 0.3
    ambiguity_threshold: 0.1
policies: null
```

**Comment by sara-tagger** [comment]
Thanks for the issue, @samsucik will get back to you about it soon!

###### You may find help in the [docs](https://rasa.com/docs/) and the [forum,](https://forum.rasa.com/) too 🤗

**Comment by melindaloubser1** [comment]
Thanks for reporting @kearnsw. What do you see in the Rasa X logs when you try to change the config file?

**Comment by kearnsw** [comment]
@melindaloubser1, I believe the error traceback I posted above is the log from the RasaX pod. It just shows the error code of 422, preceded and followed by 200 messages pinging for new models.

**Comment by melindaloubser1** [comment]
Looking at  the API spec, 422 indicates [missing mandatory keys](https://rasa.com/docs/rasa-x/pages/http-api#operation/saveModelConfig), but your config is valid - as long as there is a module `aws` installed and `ComprehendMedical` is available at that module path. 

Can you use the API directly to fetch your config? 
If so, what is the config that it returns? 
Can you use the API directly to change your config?

**Comment by kearnsw** [comment]
When I call the `/api/projects/default/settings` endpoint, I get the pre-change configuration file that is nearly equivalent to that above. 

```yaml
language: en
pipeline:
- name: WhitespaceTokenizer
- name: RegexFeaturizer
- name: LexicalSyntacticFeaturizer
- name: CountVectorsFeaturizer
- name: CountVectorsFeaturizer
  analyzer: char_wb
  min_ngram: 1
  max_ngram: 4
- name: DIETClassifier
  epochs: 100
- name: EntitySynonymMapper
- name: components.aws.ComprehendMedical
- name: FallbackClassifier
  threshold: 0.3
  ambiguity_threshold: 0.1
policies:
```

Looks like the issue is that the policies are not set, so you cannot use the default pipeline if you modify the configuration through RasaX directly. 

```json
{
    "version": "0.35.1",
    "status": "failure",
    "message": "Error: The config file is missing mandatory parameters: 'policies'. Add missing parameters to config file and try again.",
    "reason": "ConfigMissingKeys",
    "details": {},
    "help": null,
    "code": 422
}
```

**Comment by melindaloubser1** [comment]
> Looks like the issue is that the policies are not set, so you cannot use the default pipeline if you modify the configuration through RasaX directly.

Does this hold true with just `policies:` vs. `policies: null`?

**Comment by kearnsw** [comment]
Yes, both cause the same issue. So does removing the `policies` key entirely.

**Comment by kearnsw** [comment]
However, explicitly adding in the default policies did resolve the issue.

**Comment by melindaloubser1** [comment]
@degiz does sound like a bug. Should auto-config work in Rasa X when only one of policies or pipeline is provided?

**Comment by dudanogueira** [comment]
Hi, had the same problem. It also applies to pipeline

if you have pipeline: null (not sure where I got this null, I think it was from rasa-x) it will give a red alert saying:

Error
Request failed with status code 422

but if I inspect the network connection, the result is:
code: 422
details: {}
help: null
message: "Error: The config file is missing mandatory parameters: 'pipeline'. Add missing parameters to config file and try again."
reason: "ConfigMissingKeys"
status: "failure"
version: "0.35.0"

It would be nice to have a clear message stating what is wrong.

When I changed the pipelines (or policies too) to [] instead of null, it works fine.

Thanks!

**Comment by melindaloubser1** [comment]
@kearnsw 
> When I changed the pipelines (or policies too) to [] instead of null, it works fine.

Does this work for you too?

**Comment by dudanogueira** [comment]
> @kearnsw
> 
> > When I changed the pipelines (or policies too) to [] instead of null, it works fine.
> 
> Does this work for you too?

yes, it does. However i do not know where this 'null' came from.

I also noticed something similar at other parts (like domain interface). The red alert is not very clear, but once you inspect the network connection at the developer tools of the browser, you get a better message about the underlying issue.

**Comment by kearnsw** [comment]
@melindaloubser1 , that does appear to work. However, I wouldn't consider it a solution unless Rasa X by default filled that field with [] as opposed to null. Still, I think that may be confusing to the user as it isn't the way it is defined in the Rasa YAML config format.

**Comment by kearnsw** [comment]
I agree with @dudanogueira that it'd be incredibly helpful if the error messages present in the API could be shown to the user on the frontend.

**Comment by melindaloubser1** [comment]
@kearnsw very fair points.

**Comment by twerkmeister** [comment]
I successfully reproduced this issue using the rasa-x-demo repo and the following versions
```
Rasa Version     : 2.2.2
Rasa SDK Version : 2.3.1
Rasa X Version   : 0.35.1
Python Version   : 3.7.9
Operating System : Darwin-20.3.0-x86_64-i386-64bit
```
will be working on it the next days

**Comment by twerkmeister** [comment]
@kearnsw this issue will be fixed in rasa x 0.38.0! Thanks again for reporting!
